janusgraph-driver/src/main/java/org/janusgraph/graphdb/tinkerpop/io/binary/GeoshapeGraphBinarySerializer.java

```diff
@@ -40,6 +40,28 @@ public BufferInputStream(Buffer buffer) {
         public int read() {
             return buffer.readInt();
         }
+
+        @Override
+        public int read(byte[] b, int off, int len) {
+            if (b == null) {
+                throw new NullPointerException();
+            } else if (off < 0 || len < 0 || len > b.length - off) {
+                throw new IndexOutOfBoundsException();
+            } else if (len == 0) {
+                return 0;
+            }
+
+            int c = read();
+            b[off] = (byte) c;
+
+            int i = 1;
+
+            for (; i < len; i++) {
+                c = read();
+                b[off + i] = (byte) c;
+            }
+            return i;
```

Comment:
(_nitpick_) - Not necessary to resolve because this part of code was copied from InputStream implementation.

As I understand previously you could read less data then the `len` but now we always read exactly `len` amount of data. So, I guess this could be simplified a little bit.
```
for(int i=0;i<len;i++){
    b[off+i] = (byte) read();
}
return len;
```